### DISCUSSION TITLE
CombinationApprovals.VerifyAllCombinations equiavalent using Verify?

### DISCUSSION THREAD

**Discussion by ghost** [VerifyTests/Verify#546] [General]
Does the Verify library offer Combination Approvals and how would a sample one be implemented?

I would like to implement something like:

            CombinationApprovals.VerifyAllCombinations(
                DoUpdateQuality,
                new string[] { "foo", "Aged Brie", "Backstage passes to a TAFKAL80ETC concert", "Sulfuras, Hand of Ragnaros" },
                new int[] { -1, 0, 5, 10, 12 },
                new int[] { -1, 0, 49, 51 }
                );

**Comment by SimonCropp** [comment]
You can achieve this with [parameterised tests](https://github.com/VerifyTests/Verify/blob/main/docs/parameterised.md)

**Comment by StojanovicMilos** [comment]
@SimonCropp can you please elaborate further with a simple example? Let's say we want to test class Math that has a method Add that accepts two numbers and that I want to test all combinations where first number can be [1,2] while, second number can be [3,4], so it should generate 4 cases, 1+3, 1+4, 2+3 and 2+4. Thanks :)

**Reply by SimonCropp** [reply]
i added an explicit API for this scenario https://github.com/VerifyTests/Verify/blob/main/docs/combinations.md

**Reply by StojanovicMilos** [reply]
Thank you
